lib/elixir/lib/enum.ex

```diff
@@ -1284,7 +1284,7 @@ defmodule Enum do
       "2 = 4 = 6"
 
   """
-  @spec map_join(t, String.t, (element -> any)) :: String.t
+  @spec map_join(t, String.t, (element -> String.t)) :: String.t
```

Comment:
Would it be `String.t | String.Chars.t`? If it matches on `is_binary`, it passes through. `String.Chars.t` just falls through to `term`.